Src/PCompiler/CompilerCore/TypeChecker/AST/Declarations/Function.cs

```diff
@@ -70,7 +75,15 @@ public void AddLocalVariable(Variable local)
         {
             localVariables.Add(local);
         }
-
+        public void AddRequire(IPExpr expr)
```

Comment:
I remember there were no issues with pure functions where a pure function must not refer to a machine field. Can we add a static check for this?